java/client/src/org/openqa/selenium/support/ui/FluentWait.java

```diff
@@ -102,12 +107,28 @@ public FluentWait(T input, Clock clock, Sleeper sleeper) {
    * Sets how long to wait for the evaluated condition to be true. The default timeout is
    * {@link #FIVE_HUNDRED_MILLIS}.
    *
+   * @deprecated use {@link #withTimeout(long, TemporalUnit)}
+   *
    * @param duration The timeout duration.
    * @param unit The unit of time.
    * @return A self reference.
    */
+  @Deprecated
   public FluentWait<T> withTimeout(long duration, TimeUnit unit) {
-    this.timeout = new Duration(duration, unit);
+    return withTimeout(duration, toChronoUnit(unit));
+  }
+
+
+  /**
+   * Sets how long to wait for the evaluated condition to be true. The default timeout is
+   * {@link #DEFAULT_WAIT_DURATION}.
+   *
+   * @param duration The timeout duration.
+   * @param unit The unit of time.
+   * @return A self reference.
+   */
+  public FluentWait<T> withTimeout(long duration, TemporalUnit unit) {
```

Comment:
Can we please avoid changing the public APIs for now? Let's focus on removing our custom `Duration` class now.

Reply:
`java.time.Duration.of(long, TemporalUnit)` requires TemporalUnit and doesn't support TimeUnit, so this method may be `private`. But in general java 8 doesn't provide easy way of conversion between `TemporalUnit` and `TimeUnit`, so the method may be useful for people who uses `TemporalUnit` and its implementation `ChronoUnit`

Reply:
Java APIs concerning times and dates really are a mess. For now, I think we should follow the example of `java.util.concurrent` and stick with `TimeUnit`. Your `toChronoUnit` method hides this horror our users.

Reply:
I guess the alternative would be to add a `withTimeout(Duration)`, which keeps us in sync with the `Sleeper`. *sigh*